crates/core/src/writer/stats.rs

```diff
@@ -76,7 +85,21 @@ fn stats_from_file_metadata(
         .collect();
     let row_group_metadata = row_group_metadata?;
 
-    for i in 0..schema_descriptor.num_columns() {
+    let number_to_iterate = if stats_columns.is_some() {
+        schema_descriptor.num_columns()
+    } else {
+        if num_indexed_cols == -1 {
+            schema_descriptor.num_columns()
+        } else if num_indexed_cols >= 0 {
+            min(num_indexed_cols as usize, schema_descriptor.num_columns())
+        } else {
+            return Err(DeltaWriterError::DeltaTable(DeltaTableError::Generic(
+                "delta.dataSkippingNumIndexedCols valid values are >=-1".to_string(),
+            )));
+        }
+    };
```

Comment:
would it be a bit clearer if we just create the names or indexes of the columns we want to collect stats for here, rather then introducing additional branching below?

Reply:
Let me check, we need to fetch the stats by idx, so I might be able to construct a vec of indexes :)